src/sentry/models/sentryappinstallation.py

```diff
@@ -137,5 +137,29 @@ def save(self, *args, **kwargs):
     @classmethod
     def get_installed_for_org(cls, organization_id):
         return cls.objects.filter(
-            organization_id=organization_id, status=SentryAppInstallationStatus.INSTALLED
+            organization_id=organization_id,
+            status=SentryAppInstallationStatus.INSTALLED,
+            date_deleted=None,
         )
+
+    def prepare_sentry_app_components(self, component_type, project=None):
+        from sentry.coreapi import APIError
+        from sentry.mediators import sentry_app_components
+        from sentry.models import SentryAppComponent
+
+        try:
+            component = SentryAppComponent.objects.filter(
+                sentry_app_id=self.sentry_app_id, type=component_type
+            )
+
+            try:
+                sentry_app_components.Preparer.run(
+                    component=component, install=self, project=project
+                )
+                return component
+            except APIError:
+                # TODO(nisanthan): For now, skip showing the UI Component if the API requests fail
+                return None
+
+        except SentryAppComponent.DoesNotExist:
```

Comment:
does this line catch something in `sentry_app_components.Preparer.run`? because `SentryAppComponent.objects.filter` will just return 0 elements instead of raising this error, (`.get` would). if this doesn't catch anything in `sentry_app_components.Preparer.run`, you could also move it right after the component line as to not have nested trys. if it does, you could move it into the same exception as `APIError` and get rid of the first try.

Reply:
ahh good catch! its supposed to be a `.get` query

Reply:
👌🏼